### DISCUSSION TITLE
How to make use of "code" cli command in postAttachCommand script

### DISCUSSION THREAD

**Discussion by jota0222** [devcontainers/community#94] [Q&A] [answered]
I would like to open a file using `code -r` or open an url in the host browser using `code --openExternal` from the `postAttachCommand`, `postStartCommand` or the `postCreatedCommand`, but it seems not all the commands are available in this part of the lifecycle because it fails with this message:

![imagen](https://github.com/devcontainers/community/assets/12126375/8d2a3e49-8d86-4bdc-9910-36cd13cc9715)

However, it works as expected if I use it from the terminal. I must mention I also tried with the `$BROWSER` variable and it seems it's not defined when the lifecycles are running.

Do you have any way I can use to run those commands in the lifecycle hooks?

**Comment by eljog** [comment, marked as answer]
Thank you for opening this discussion.

The lifecycle commands such as the `onCreateCommand` (the one you tried), `postStartCommand` and `postAttachCommand` are run non-interactively, hence it won't be able to launch a UI action.
If you are using Codespaces you could use `postAttachCommand` that can run interactive commands such as `code --openExternal https://www.microsoft.com`.

However it appears that it is not supported in vscode remote/devcontainers extension. If you are using this extension you may open a request at https://github.com/microsoft/vscode-remote-release/issues
